# GWAS Locus Browser Coding Variants Script
- **Author** - Frank Grenn and Cornelis Blauwendraat
- **Date Started** - December 2019
- **Quick Description:** scripts to get coding variant data including CADD scores, LD values and frequencies.

## Layout
```
|-- AppDataProcessing
|   |-- genes_by_locus.csv
|   |-- codingvars
|   |   |-- get_05tags_for_META5_HRC.sh
|   |   |-- annotation_of_tags.sh
|   |   |-- annotated_R05_tags_META5.txt
|   |   |-- CodingVariant.csv
|   |   |-- LD
|   |   |   |-- out
|   |   |   |   |-- 6:30108683_6:30139699.log
|   |   |   |   |-- 20:6006041_20:6011934.log
|   |   |   |   |-- ...
|   |   |   |-- CalculateLD.ipynb
|   |   |   |-- calculateLD.swarm
|   |   |   |-- Variants.csv
|   |   |   |-- ranges.txt
|   |   |-- get_CADD_annovar.sh
|   |   |-- ReadCodingVariantData.py
|   |   |-- get_frequencies_annovar.sh
|   |   |-- codingvars.avinput
|   |   |-- codingvars.avinput.hg19_cadd_dropped
|   |   |-- codingvars.avinput.hg19_cadd_filtered
|   |   |-- codingvars.avinput.hg19_gnomad_genome_dropped
|   |   |-- codingvars.avinput.hg19_gnomad_genome_filtered
|   |   |-- ...
|   |-- results
|   |   |-- CodingVariants.csv
|   |   |-- CodingVariantLD.csv
```

---

## 1) Generate `CodingVariant.csv` File (skip if the file exists)

### Run
* on biowulf
* run code in `get_05tags_for_META5_HRC.sh` followed by `annotation_of_tags.sh`
* may still need to manipulate output afterwards

### Input
* see scripts

### Output
* a `annotated_R05_tags_META5.txt` that can be filtered to create the `CodingVariant.csv` needed in the next steps

---

## 2) Coding Variant LD Data

### Run
* on biowulf
* run through `LD/CalculateLD.ipynb`
  * the notebook will also generate a `calculateLD.swarm` to run (see the notebook)


### Input
* a `Variants.csv` file containing a `SNP`, `CHR`, `BP` and `Locus Number` column for each variant of interest (meta5 + progression)
* a `ranges.txt` file containing chr, bp position - 1MB, bp position + 1MB, and an id for each variant of interest. 
* a `CodingVariant.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each coding variant of interest
* plink binary files for the gwas data

### Output
* a `calculateLD.swarm` file that will use plink to get LD values for all the variants
* many `.log` files in the `out` directory generated by the notebook and swarm file containing the LD values
* a `CodingVariantLD.csv` to the `results` directory containing all the coding variant's LD values in one file

---

## 3) Coding Variant CADD Scores

### Run
* on biowulf 
* `sbatch get_CADD_annovar.sh`

### Input
* a `CodingVariant.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each coding variant of interest

### Output
* a `codingvars.avinput` file that is formatted for the annovar command
* a `codingvars.avinput.hg19_cadd_dropped` containing the CADD scores we want
* a `codingvars.avinput.hg19_cadd_filtered` containing the data we don't need

---

## 4) Coding Variant frequencies

### Run
* on biowulf 
* `sbatch get_frequencies_annovar.sh`

### Input
* a `CodingVariant.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each coding variant of interest

### Output
* a `codingvars.avinput` file that is formatted for the annovar command
* a `codingvars.avinput.hg19_gnomad_genome_dropped` containing the frequencies we want
* a `codingvars.avinput.hg19_gnomad_genome_filtered` containing the data we don't need

---

## 5) Combine Coding Variant Data (combine data from (2) and (3) into one file)

### Run
* on biowulf 
* `python codingvars/ReadCodingVariantData.py`

### Input
* a `CodingVariant.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each coding variant of interest
* the `codingvars.avinput.hg19_gnomad_genome_dropped` from (3)
* the `codingvars.avinput.hg19_cadd_dropped` from (2)

### Output
* a `CodingVariants.csv` file in the `results` folder containing all of the coding variants with their CADD and frequency data obtained using annovar 