# GWAS Locus Browser Associated Variant Phenotypes Scripts
- **Author** - Frank Grenn and Cornelis Blauwendraat
- **Date Started** - December 2019
- **Quick Description:** scripts to get associated variant phenotype data including LD values and frequencies.
- **Data:** - (https://www.ebi.ac.uk/gwas/docs/file-downloads)[GWAS Catalog]

## Layout
```
|-- AppDataProcessing
|   |-- genes_by_locus.csv
|   |-- phenovars
|   |   |-- Variant in other GWAS - Sheet1.csv
|   |   |-- LD
|   |   |   |-- out
|   |   |   |   |-- 6:30108683_6:30139699.log
|   |   |   |   |-- 20:6006041_20:6011934.log
|   |   |   |   |-- ...
|   |   |   |-- CalculateLD.ipynb
|   |   |   |-- calculateLD.swarm
|   |   |   |-- Variants.csv
|   |   |   |-- ranges.txt
|   |   |-- ReadPhenoVariantData.py
|   |   |-- get_frequencies_annovar.sh
|   |   |-- phenovars.avinput
|   |   |-- phenovars.avinput.hg19_gnomad_genome_dropped
|   |   |-- phenovars.avinput.hg19_gnomad_genome_filtered
|   |   |-- ...
|   |-- results
|   |   |-- PhenotypeVariantData.csv
|   |   |-- PhenotypeVariantLD.csv
```

---

## 1) Phenotype Variant LD Data

### Run
* on biowulf
* run through `LD/CalculateLD.ipynb`
  * the notebook will also generate a `calculateLD.swarm` to run (see the notebook)


### Input
* a `Variants.csv` file containing a `SNP`, `CHR`, `BP` and `Locus Number` column for each variant of interest (meta5 + progression)
* a `ranges.txt` file containing chr, bp position - 1MB, bp position + 1MB, and an id for each variant of interest. 
* a `Variant in other GWAS - Sheet1.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each phenotype variant of interest
* plink binary files for the gwas data
### Output
* a `calculateLD.swarm` file that will use plink to get LD values for all the variants
* many `.log` files in the `out` directory generated by the notebook and swarm file containing the LD values
* a `PhenotypeVariantLD.csv` to the `results` directory containing all the phenotype variant's LD values in one file

---

## 2) Get frequency data using ANNOVAR

### Run
* on biowulf 
* `sbatch get_frequencies_annovar.sh`

### Input
* a `Variant in other GWAS - Sheet1.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each phenotype variant of interest

### Output
* a `phenovars.avinput` file that is formatted for the annovar command
* a `phenovars.avinput.hg19_gnomad_genome_dropped` containing the frequencies we want
* a `phenovars.avinput.hg19_gnomad_genome_filtered` containing the data we don't need

---

## 3) Combine Phenotype Variant Data

### Run
* on biowulf 
* `python phenovars/ReadPhenoVariantData.py`

### Input
* a `Variant in other GWAS - Sheet1.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene, for each phenotype variant of interest
* the `GWAS_catalogv1.0.2-associations.txt` to obtain PMIDs
* the `phenovars.avinput.hg19_gnomad_genome_dropped` from (2) containing frequency data

### Output
* a `PhenotypeVariantData.csv` file in the `results` folder containing all of the phenotype variants with their frequency data and PMIDs formatted as hyperlinks