{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Calculate Phenotype Variant LD Values\n",
    "Frank Grenn  \n",
    "January 2020\n",
    "\n",
    "### Input\n",
    "* a `Variants.csv` file containing a `SNP`, `CHR`, `BP` and `Locus Number` column for each variant of interest (meta5 + progression)\n",
    "* a `ranges.txt` file containing chr, bp position - 1MB, bp position + 1MB, and an id for each variant of interest. \n",
    "* a `Variant in other GWAS - Sheet1.csv` file containing the rsid, chr, bp, locus number, associated disease, amino acid change, and gene for each phenotype variant of interest\n",
    "* plink binary files for the gwas data\n",
    "\n",
    "### Output\n",
    "* a `calculateLD.swarm` file that will use plink to get LD values for all the variants\n",
    "* many `.log` files in the `out` directory generated by the notebook and swarm file containing the LD values\n",
    "* a `PhenotypeVariantLD.csv` to the `results` directory containing all the phenotype variant's LD values in one file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import subprocess\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SNP</th>\n",
       "      <th>CHR</th>\n",
       "      <th>BP</th>\n",
       "      <th>Locus Number</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>rs114138760</td>\n",
       "      <td>1</td>\n",
       "      <td>154898185</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>rs35749011</td>\n",
       "      <td>1</td>\n",
       "      <td>155135036</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>rs76763715</td>\n",
       "      <td>1</td>\n",
       "      <td>155205634</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>rs6658353</td>\n",
       "      <td>1</td>\n",
       "      <td>161469054</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>rs11578699</td>\n",
       "      <td>1</td>\n",
       "      <td>171719769</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>rs55818311</td>\n",
       "      <td>19</td>\n",
       "      <td>2341047</td>\n",
       "      <td>76</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>rs77351827</td>\n",
       "      <td>20</td>\n",
       "      <td>6006041</td>\n",
       "      <td>77</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>rs2248244</td>\n",
       "      <td>21</td>\n",
       "      <td>38852361</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>rs61863020</td>\n",
       "      <td>10</td>\n",
       "      <td>112956055</td>\n",
       "      <td>prog1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>rs382940</td>\n",
       "      <td>9</td>\n",
       "      <td>108058562</td>\n",
       "      <td>prog2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            SNP  CHR         BP Locus Number\n",
       "0   rs114138760    1  154898185            1\n",
       "1    rs35749011    1  155135036            1\n",
       "2    rs76763715    1  155205634            1\n",
       "3     rs6658353    1  161469054            2\n",
       "4    rs11578699    1  171719769            3\n",
       "..          ...  ...        ...          ...\n",
       "87   rs55818311   19    2341047           76\n",
       "88   rs77351827   20    6006041           77\n",
       "89    rs2248244   21   38852361           78\n",
       "90   rs61863020   10  112956055        prog1\n",
       "91     rs382940    9  108058562        prog2\n",
       "\n",
       "[92 rows x 4 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_csv(\"$PATH1/phenovars/LD/Variants.csv\")\n",
    "variants = variants.astype({'Locus Number':'str'})\n",
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "375\n",
      "          ID      location locus number  Chr     Start       End Ref Alt  \\\n",
      "0  rs3104413  543050:10:00           31    6  32582650  32582650   C   G   \n",
      "1  rs3957146  544698:10:00           31    6  32681530  32681530   T   C   \n",
      "2  rs3998159  544706:19:00           31    6  32682019  32682019   A   C   \n",
      "3  rs6931277  543061:57:00           31    6  32583357  32583357   A   T   \n",
      "4  rs7454108  544697:23:00           31    6  32681483  32681483   T   C   \n",
      "\n",
      "  Func.refGene       Gene.refGene     GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "0   intergenic  HLA-DRB1;HLA-DQA1  dist=25037;dist=22533                  .   \n",
      "1   intergenic  HLA-DQB1;HLA-DQA2  dist=47064;dist=27633                  .   \n",
      "2   intergenic  HLA-DQB1;HLA-DQA2  dist=47553;dist=27144                  .   \n",
      "3   intergenic  HLA-DRB1;HLA-DQA1  dist=25744;dist=21826                  .   \n",
      "4   intergenic  HLA-DQB1;HLA-DQA2  dist=47017;dist=27680                  .   \n",
      "\n",
      "  AAChange.refGene                           other associated disease  \n",
      "0                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "1                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "2                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "3                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "4                .  ACPA-positive rheumatoid arthritis (smoking in...  \n"
     ]
    }
   ],
   "source": [
    "phenovars = pd.read_csv(\"$PATH1/phenovars/Variant in other GWAS - Sheet1.csv\")\n",
    "phenovars = phenovars.astype({'locus number':'str'})\n",
    "print(len(phenovars.index))\n",
    "print(phenovars.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "370\n",
      "          ID      location locus number  Chr     Start       End Ref Alt  \\\n",
      "0  rs3104413  543050:10:00           31    6  32582650  32582650   C   G   \n",
      "1  rs3957146  544698:10:00           31    6  32681530  32681530   T   C   \n",
      "2  rs3998159  544706:19:00           31    6  32682019  32682019   A   C   \n",
      "3  rs6931277  543061:57:00           31    6  32583357  32583357   A   T   \n",
      "4  rs7454108  544697:23:00           31    6  32681483  32681483   T   C   \n",
      "\n",
      "  Func.refGene       Gene.refGene     GeneDetail.refGene ExonicFunc.refGene  \\\n",
      "0   intergenic  HLA-DRB1;HLA-DQA1  dist=25037;dist=22533                  .   \n",
      "1   intergenic  HLA-DQB1;HLA-DQA2  dist=47064;dist=27633                  .   \n",
      "2   intergenic  HLA-DQB1;HLA-DQA2  dist=47553;dist=27144                  .   \n",
      "3   intergenic  HLA-DRB1;HLA-DQA1  dist=25744;dist=21826                  .   \n",
      "4   intergenic  HLA-DQB1;HLA-DQA2  dist=47017;dist=27680                  .   \n",
      "\n",
      "  AAChange.refGene                           other associated disease  \n",
      "0                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "1                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "2                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "3                .  ACPA-positive rheumatoid arthritis (smoking in...  \n",
      "4                .  ACPA-positive rheumatoid arthritis (smoking in...  \n"
     ]
    }
   ],
   "source": [
    "#drop identical rows\n",
    "phenovars = phenovars.drop_duplicates()\n",
    "print(len(phenovars.index))\n",
    "print(phenovars.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "mkdir $PATH1/phenovars/LD/out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in range(len(variants.index)):\n",
    "    locus = variants.iloc[i]['Locus Number']\n",
    "    #print(locus)\n",
    "    \n",
    "    locus_phenovars = phenovars.loc[phenovars['locus number'] == locus]\n",
    "    #print(locus_phenovars)\n",
    "    \n",
    "    snp1 = str(variants.iloc[i]['CHR']) + \":\" + str(variants.iloc[i]['BP'])\n",
    "\n",
    "    if(len(locus_phenovars.index)!=0):\n",
    "        for i in range(len(locus_phenovars.index)):\n",
    "\n",
    "            snp2 = str(locus_phenovars.iloc[i]['Chr']) + \":\" + str(locus_phenovars.iloc[i]['Start'])\n",
    "\n",
    "\n",
    "\n",
    "            with open(\"$PATH1/phenovars/LD/calculateLD.swarm\", 'a') as outfile:\n",
    "                outfile.write(\"plink --bfile $PATH2/HARDCALLS_PD_september_2018_no_cousins --ld {} {} --extract range $PATH1/phenovars/LD/ranges.txt --out {}_{}\\n\".format(snp1,snp2,snp1,snp2))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run the swarm file\n",
    "#cd $PATH1/phenovars/LD/out\n",
    "#swarm -f $PATH1/phenovars/LD/calculateLD.swarm --partition quick --module plink"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "447\n",
      "         rsid1         snp1       rsid2         snp2           r2     dprime\n",
      "0  rs114138760  1:154898185  rs12726330  1:155108167  0.000197778          1\n",
      "1  rs114138760  1:154898185   rs2230288  1:155206167  0.000183319          1\n",
      "2  rs114138760  1:154898185  rs35749011  1:155135036  0.000185332          1\n",
      "3   rs35749011  1:155135036  rs12726330  1:155108167     0.929883   0.996158\n",
      "4   rs35749011  1:155135036   rs2230288  1:155206167     0.937428   0.973508\n",
      "         rsid1         snp1       rsid2         snp2        r2     dprime\n",
      "442  rs8087969  18:48683589   rs8088085  18:48708548  0.977947   0.990706\n",
      "443  rs8087969  18:48683589  rs16952939  18:48743642  0.923426   0.969977\n",
      "444  rs8087969  18:48683589   rs1941961  18:48744150  0.672056   0.946768\n",
      "445  rs8087969  18:48683589  rs62092949  18:48779673  0.718631   0.908091\n",
      "446  rs8087969  18:48683589  rs57976196  18:48738645  0.921946   0.971478\n"
     ]
    }
   ],
   "source": [
    "#read stuff\n",
    "df = pd.DataFrame(columns=['rsid1','snp1','rsid2','snp2','r2','dprime'])\n",
    "for i in range(len(variants.index)):\n",
    "    locus = variants.iloc[i]['Locus Number']\n",
    "    #print(locus)\n",
    "    \n",
    "    locus_phenovars = phenovars.loc[phenovars['locus number'] == locus]\n",
    "    #print(locus_phenovars)\n",
    "    \n",
    "    snp1 = str(variants.iloc[i]['CHR']) + \":\" + str(variants.iloc[i]['BP'])\n",
    "    rsid1 = variants.iloc[i]['SNP']\n",
    "    if(len(locus_phenovars.index)!=0):\n",
    "        for i in range(len(locus_phenovars.index)):\n",
    "            #reset the read string to null\n",
    "            dataline='null'\n",
    "            snp2 = str(locus_phenovars.iloc[i]['Chr']) + \":\" + str(locus_phenovars.iloc[i]['Start'])\n",
    "            rsid2 = locus_phenovars.iloc[i]['ID']\n",
    "            #print(\"{} {}\".format(snp1, snp2))\n",
    "\n",
    "\n",
    "            \n",
    "            file = open(\"$PATH1/phenovars/LD/out/\"+str(snp1)+\"_\"+str(snp2)+\".log\",\"r\")\n",
    "            \n",
    "            for line in file:\n",
    "                if re.search(\"R-sq\", line):\n",
    "                    dataline = line\n",
    "                    break\n",
    "            \n",
    "            #only add new data if 'R-sq' was found (meaning there was data in the log file and the 'null' value assigned earlier was overwritten)\n",
    "            if(dataline!='null'):\n",
    "                #mess with the strings\n",
    "                dataline = dataline.strip('R-sq = ')\n",
    "                dataline = dataline.strip(' ')\n",
    "                splitdata = dataline.split(\"D' =\")\n",
    "                Rsq = splitdata[0]\n",
    "                dprime = splitdata[1]\n",
    "            \n",
    "                df = df.append({'rsid1': rsid1,'snp1': snp1,'rsid2':rsid2, 'snp2': snp2, 'r2':Rsq.strip(' '), 'dprime':dprime.strip('\\n')}, ignore_index = True)\n",
    "        \n",
    "print(len(df.index))\n",
    "print(df.head())\n",
    "print(df.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "df.to_csv(\"$PATH1/results/PhenotypeVariantLD.csv\", index = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
